### PR TITLE
Implemented xep-0352: Client State Indication

### PR DESCRIPTION
**Pull request by fbeutel**

### CODE REVIEW

File: src/base/QXmppStreamFeatures.cpp [hunk 1]
```diff
@@ -94,6 +94,16 @@ void QXmppStreamFeatures::setTlsMode(QXmppStreamFeatures::Mode mode)
     m_tlsMode = mode;
 }
```
Comment (jlaine) [inline src/base/QXmppStreamFeatures.cpp]: Please document these two members

File: src/client/QXmppClient.cpp [hunk 2]
```diff
@@ -320,6 +324,27 @@ bool QXmppClient::isConnected() const
     return d->stream->isConnected();
 }
 
+/// Returns true if the current client state is "active", false if it is "inactive"
+/// See xep-0352.
+
+bool QXmppClient::clientState()
```
Comment (jlaine) [inline src/client/QXmppClient.cpp]: I am not sure about the member name, or using a boolean to convey this information. Code like this does not seem very readable:

if (client->clientState()) {
..
}

**Comment by renini** [comment]
+1 will this get merged into master?

**Comment by jlaine** [comment]
Superseed by PR #159

**Pull request closed without merging**